mysql/datadog_checks/mysql/execution_plans.py

```diff
@@ -1,72 +1,47 @@
 import json
+import time
 from contextlib import closing
 
-import pymysql
-
 import datadog_agent
+import pymysql
+from datadog import statsd
 from datadog_checks.base import is_affirmative
-
-from datadog_checks.base.utils.db.statement_metrics import is_dbm_enabled
 from datadog_checks.base.utils.db.sql import compute_sql_signature, compute_exec_plan_signature, submit_exec_plan_events
-
+from datadog_checks.base.utils.db.statement_metrics import is_dbm_enabled
+from datadog_checks.base.utils.db.utils import ConstantRateLimiter, ExpiringCache
 
 VALID_EXPLAIN_STATEMENTS = frozenset({
-  'select',
-  'table',
-  'delete',
-  'insert',
-  'replace',
-  'update',
+    'select',
+    'table',
+    'delete',
+    'insert',
+    'replace',
+    'update',
 })
 
+# default sampling settings for events_statements_* tables
+# rate limit is in samples/second
+# {table -> rate-limit}
+DEFAULT_EVENTS_STATEMENTS_RATE_LIMITS = {
+    'events_statements_history_long': 0,
+    'events_statements_history': 1,
+    'events_statements_current': 10,
+}
+
 
 class ExecutionPlansMixin(object):
     """
     Mixin for collecting execution plans from query samples.
     """
 
     def __init__(self, *args, **kwargs):
-        # TODO: Make this a configurable limit
-        self.query_limit = 500
-        self._checkpoint = None
-        self._auto_enable_eshl = None
-    
-    def _enable_performance_schema_consumers(self, db):
-        query = """UPDATE performance_schema.setup_consumers SET enabled = 'YES' WHERE name = 'events_statements_history_long'"""
-        with closing(db.cursor()) as cursor:
-            try:
-                cursor.execute(query)
-            except pymysql.err.OperationalError as e:
-                if e.args[0] == 1142:
-                    self.log.error('Unable to create performance_schema consumers: %s', e.args[1])
-                else:
-                    raise
-            except pymysql.err.InternalError as e:
-                if e.args[0] == 1290:
-                    self.log.warning('Unable to create performance_schema consumers because the instance is read-only')
-                    self._auto_enable_eshl = False
-                else:
-                    raise
-            else:
-                self.log.info('Successfully enabled events_statements_history_long consumers')
+        # checkpoint at zero so we pull the whole history table on the first run
+        self._checkpoint = 0
+        self._expiring_cache = ExpiringCache()
```

Comment:
small nit: Use a more descriptive name for the attribute, e.g. `_collection_strategy_cache`

Reply:
but we're going to use it to cache other stuff, like your upcoming changes to cache non-retryable explain errors?